### PR TITLE
fix: Remove indentation from statistics logging and print the data in tables

### PR DESCRIPTION
**Pull request by TymeeK**
### Description

The purpose of the PR is to fix the indentation of statistics logging. It was originally 8 space indentation but now it is changed to be all on one line.

### Issues

This fixes issue #306

### Testing

I went through the unit test files and logged out what the output would be for the statistics loggin.

Example: 
```
[crawlee.basic_crawler.basic_crawler] INFO  Final request statistics: {"requests_finished": 3, "requests_failed": 0, "retry_histogram": [3], "request_avg_failed_duration": null, "request_avg_finished_duration": 0.000984, "requests_finished_per_minute": 7348, "requests_failed_per_minute": 0, "request_total_duration": 0.002951, "requests_total": 3, "crawler_runtime": 0.024498}
```

### Checklist

- [ ] Changes are described in the `CHANGELOG.md`
- [x] CI passed

### CODE REVIEW

**Comment by B4nan** [comment]
FYI we had a quick poll about this and in the end we agreed we want to print this as a formatted table instead of JSON, so we won't be merging this one.

Something like the following (but with snake_case instead of camelCase - this snippet was build based on the JS version)

```
┌──────────────────────────────────┬────────┐
│ requestsFinished                 │ 59     │
│ requestsFailed                   │ 0      │
│ retryHistogram                   │ [ 59 ] │
│ requestAvgFailedDurationMillis   │ null   │
│ requestAvgFinishedDurationMillis │ 1071   │
│ requestsFinishedPerMinute        │ 67     │
│ requestsFailedPerMinute          │ 0      │
│ requestTotalDurationMillis       │ 63215  │
│ requestsTotal                    │ 59     │
│ crawlerRuntimeMillis             │ 53197  │
└──────────────────────────────────┴────────┘
```

**Comment by janbuchar** [comment]
We should be able to print tables using `rich.Table` - see https://rich.readthedocs.io/en/stable/tables.html

**Comment by TymeeK** [comment]
Thanks for your feedback! I am brand new to open source. I did make some furthers changes by using a table to print it out as a formatted table. Should I open a new PR or update this one?

**Comment by janbuchar** [comment]
> Thanks for your feedback! I am brand new to open source. I did make some furthers changes by using a table to print it out as a formatted table. Should I open a new PR or update this one?

Awesome! Just commit the changes you made and push it to your fork and the PR will be updated automatically

File: src/crawlee/basic_crawler/basic_crawler.py [hunk 1]
```diff
@@ -379,7 +379,8 @@ def sigint_handler() -> None:
         self._has_finished_before = True
 
         final_statistics = self._statistics.calculate()
-        self._logger.info(f'Final request statistics: {final_statistics}')
+        self._logger.info('Final request statistics: ')
+        final_statistics.to_table()
```
Comment (janbuchar) [inline src/crawlee/basic_crawler/basic_crawler.py:383]: The table should be concatenated with the logged message. What the `self._logger.info` call does depends on the configuration of the `logging` system, which is open to library users. Therefore, the two parts of the log line may end up in different places.

File: src/crawlee/statistics/models.py [hunk 2]
```diff
@@ -27,6 +29,20 @@ class FinalStatistics:
     requests_total: int
     crawler_runtime: timedelta
 
+    def to_table(self) -> None:
+        """Print out the Final Statistics data as a table."""
+        table = Table()
+        table.add_column('INFO')
+        table.add_column('Statistics')
```
Comment (janbuchar) [inline src/crawlee/statistics/models.py:36]:
```suggestion
        table = Table(show_header=False)
        table.add_column()
        table.add_column()
```

The headers are not particularly useful in this case.

**Review by janbuchar — CHANGES_REQUESTED** [review]

**Review by vdusek — APPROVED** [review]
LGTM

**Pull request merged**